src/views/student/Immatriculation.vue

```diff
@@ -0,0 +1,230 @@
+                                class="mt-4"
+                                :elements="selectedFieldsOfStudy"
+                                @on-remove="removeFieldOfStudy"
+                            />
+                            <span v-else class="input-label-warning">Please select at least one field of study.</span>
+                        </div>
+                    </div>
+                </div>
+                <p v-if="errorBag.hasNested('matriculation')" class="error-message">{{ errorBag.getNested("matriculation") }}</p>
+            </div>
+        </base-section>
+        <button-section>
+            <template #right>
+                <button id="addImmatriculationData" :disabled="!validSelection" class="btn w-full lg:w-64" @click="updateImmatriculation">
+                    Update
+                </button>
+            </template>
+        </button-section>
+    </base-view>
+</template>
+
+<script lang="ts">
+    import { computed, onBeforeMount, ref } from "vue";
+    import SubjectMatriculation from "@/api/api_models/matriculation_management/SubjectMatriculation";
+    import GenericResponseHandler from "@/use/helpers/GenericResponseHandler";
+    import LoadingSpinner from "@/components/common/loading/Spinner.vue";
+    import ImmatriculationHistory from "@/components/common/immatriculation/ImmatriculationHistory.vue";
+    import ErrorBag from "@/use/helpers/ErrorBag";
+    import { useStore } from "@/use/store/store";
+    import CertificateManagement from "@/api/CertificateManagement";
+    import { useToast } from "@/toast";
+    import { updateMatriculation } from "@/api/abstractions/FrontendSigning";
+    import BaseView from "@/views/common/BaseView.vue";
+    import BaseSection from "@/components/common/section/BaseSection.vue";
+    import SectionHeader from "@/components/common/section/SectionHeader.vue";
+    import Select from "@/components/common/Select.vue";
+    import ButtonSection from "@/components/common/section/ButtonSection.vue";
+    import SearchableSelect from "@/components/common/SearchableSelect.vue";
+    import TagList from "@/components/common/TagList.vue";
+    import SearchSelectOption from "@/use/helpers/SearchSelectOption";
+    import ExaminationRegulationManagement from "@/api/ExaminationRegulationManagement";
+
+    export default {
+        components: {
+            ButtonSection,
+            Select,
+            SectionHeader,
+            LoadingSpinner,
+            ImmatriculationHistory,
+            BaseView,
+            BaseSection,
+            TagList,
+            SearchableSelect,
+        },
+        setup() {
+            let refreshKey = ref(false);
+            let isLoading = ref(true);
+            let fieldsOfStudy = ref(["Computer Science", "Physics", "Mechanical Engineering"] as string[]);
+            let semesterType = ref("");
+            let year = ref("");
+            let selectedFieldsOfStudy = ref([] as string[]);
```

Comment:
```suggestion
            const refreshKey = ref(false);
            const isLoading = ref(true);
            const fieldsOfStudy = ref([] as string[]);
            const semesterType = ref("");
            const year = ref("");
            const selectedFieldsOfStudy = ref([] as string[]);
```
pretty sure thats debug code